java/src/main/kotlin/io/spine/validation/java/generate/option/bound/RangeGenerator.kt

```diff
@@ -116,6 +117,27 @@ private class GenerateRange(
         FIELD_VALUE to fieldType.stringValueOf(fieldValue),
         FIELD_TYPE to StringLiteral(fieldType.name),
         PARENT_TYPE to typeName,
-        RANGE_VALUE to StringLiteral(view.range)
+        RANGE_VALUE to withFieldValue()
     )
+
+    /**
+     * Constructs a string [Expression] for the range literal, inserting bound
+     * field values, if any.
+     *
+     * The method splits a range string into its lower and upper parts, appending any
+     * referenced field’s actual values in parentheses, and then concatenates back
+     * these parts using the original delimiter.
+     */
+    private fun withFieldValue(): Expression<String> {
+        val (left, right) = view.range.split(DELIMITER).map { it.trim() }
+        val leftValue = left.withFieldValue(lower)
+        val rightBrace = right.last().toString()
+        val rightValue = right.dropLast(1).withFieldValue(upper) + rightBrace
+        return leftValue + " $DELIMITER " + rightValue
+    }
 }
+
+/**
+ * The range delimiter.
+ */
+private const val DELIMITER = ".."
```

Comment:
This constant hangs in a „vacuum“. Don't we have a proper outer class or object for it?